session/session.go

```diff
@@ -540,6 +541,22 @@ func (s *session) CommitTxn(ctx context.Context) error {
 
 	var commitDetail *execdetails.CommitDetails
 	ctx = context.WithValue(ctx, execdetails.CommitDetailCtxKey, &commitDetail)
+	is := infoschema.GetInfoSchema(s)
+	txnScope := s.GetSessionVars().TxnScope
+	for physicalTableID := range s.sessionVars.TxnCtx.TableDeltaMap {
+		bundle, ok := is.RuleBundles()[placement.GroupID(physicalTableID)]
+		if !ok {
+			continue
+		}
+		// FIXME: currently we assumes "zone" is the dcLabel key in Store
+		dcLocation, ok := placement.GetLeaderDCByBundle(bundle, "zone")
+		if !ok {
+			continue
```

Comment:
It should report an error here. If the partition has no placement rules, it may be located anywhere. So local transactions shouldn't visit them.

Reply:
updated.